fendermint/vm/topdown/src/lib.rs

```diff
@@ -18,9 +36,35 @@ pub struct IPCParentFinality {
     pub height: BlockHeight,
     /// The block hash. For FVM, it is a Cid. For Evm, it is bytes32 as one can now potentially
     /// deploy a subnet on EVM.
-    pub block_hash: Vec<u8>,
-    /// new top-down messages finalized in this PoF
-    pub top_down_msgs: Vec<CrossMsg>,
-    /// latest validator configuration information from the parent.
-    pub validator_set: ValidatorSet,
+    pub block_hash: BlockHash,
+}
+
+pub trait ParentViewProvider {
```

Comment:
Do you mind adding a few comments explaining what `ParentViewProvider` and `ParentFinalityProvider` are?  The former is the local view of the current finality  of the parent seen by the peer; while the latter handles the finality in the parent that the validators in the child subnet have already agreed upon. I think explicitly stating the distinction can be useful.

Reply:
added